machine_learning/neural_network.cpp

```diff
@@ -0,0 +1,719 @@
+                    }
+                    return details;
+                }
+            public:
+                /**
+                 * Default Constructor for class NeuralNetwork. This constructor
+                 * is used to create empty variable of type NeuralNetwork class.
+                 */   
+                NeuralNetwork() = default;
+                /**
+                 * Constructor for class NeuralNetwork. This constructor
+                 * is used by user.
+                 * @param config vector containing pair (neurons, activation)
+                 */   
+                explicit NeuralNetwork(const std::vector <std::pair<int, std::string>> &config) {
+                    // First layer should not have activation
+                    if(config.begin() -> second != "none") {
+                        std::cerr << "ERROR: First layer can't have activation other than none";
+                        std::cerr << std::endl;
+                        std::exit(EXIT_FAILURE);
+                    }
+                    // Network should have atleast two layers
+                    if(config.size() <= 1) {
+                        std::cerr << "ERROR: Invalid size of network, ";
+                        std::cerr << "Atleast two layers are required";
+                        std::exit(EXIT_FAILURE);
+                    }
+                    // Sepratly creating first layer so it can have unit matrix 
+                    // as kernal.
+                    layers.push_back(Layer(config[0].first, 
+                                           config[0].second, 
+                                           {config[0].first, config[0].first},
+                                           false));
+                    // Creating remaining layers
+                    for(size_t i = 1; i < config.size(); i++) {
+                        layers.push_back(Layer(config[i].first, 
+                                               config[i].second,
+                                               {config[i - 1].first, config[i].first},
+                                               true));
+                    }
+                    std::cout << "INFO: Network constructed successfully" << std::endl;
+                }
+
+                /**
+                 * Function to get X and Y from csv file (where X = data, Y = label)
+                 * @param file_name csv file name
+                 * @param last_label flag for whether label is in first or last column
+                 * @param normalize flag for whether to normalize data 
+                 * @return returns pair of X and Y
+                 */  
+                std::pair<std::vector<std::vector<std::valarray<double>>>, std::vector<std::vector<std::valarray<double>>>>
+                get_XY_from_csv(const std::string &file_name, 
+                                     const bool &last_label, 
+                                     const bool &normalize) {
```

Comment:
```suggestion
                                     const bool &normalize, const int slip_lines=0) {
```
With the suggestion made above for CSV file, an optional parameter to skip header lines in a CSV file should help. In the beginning of the code, after opening the file, these lines can be skipped as you did on line# 331.